### ISSUE TITLE
S4261: FP on methods which return IAsyncEnumerable<T> 

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#3433]
### Description

Issues are raised for methods which end with "Async" and return `IAsyncEnumerable<T>`.

### Repro steps

```
using System;
using System.Collections.Generic;

namespace Tests.Diagnostics
{
    public abstract class IAsyncEnumerableReproducer
    {
        protected abstract IAsyncEnumerable<int> GetFilesAsync(string folderName); // Noncompliant - FP, async method
    }
}
```

### Expected behavior

Issue is not raised for methods which end with "Async" and return `IAsyncEnumerable<T>`.

### Actual behavior

Since the rule does not consider `IAsyncEnumerable<T>` return type, it will raise a FP in this case.

### Related information

* C#/VB.NET Plugins Version: https://github.com/SonarSource/sonar-dotnet/releases/tag/8.9.0.19135
* Source: https://community.sonarsource.com/t/rule-s4261-net-core-3-1-false-positive-on-iasyncenumerable/27326
* Rule: https://rules.sonarsource.com/csharp/RSPEC-4261

**Comment by lg2de** [comment]
Wondering why #3264 with similar request (?) was rejected...
